src/Microsoft.OData.Client/ClientEdmModel.cs

```diff
@@ -328,7 +328,7 @@ private static Type[] GetTypeHierarchy(Type type, out PropertyInfo[] keyProperti
                 {
                     hierarchy.Insert(0, type);
                 }
-                while ((type = c.PlatformHelper.GetBaseType(type)) != null && ClientTypeUtil.GetPropertiesOnType(type, false /*declaredOnly*/).Any());
+                while ((type = c.PlatformHelper.GetBaseType(type)) != null );

```

Comment:
You'll want to update the comment on L326 as well otherwise it might be very confusing for someone reading this file later on.
Also add some tests for this